routing/pathfind_test.go

```diff
@@ -620,56 +628,86 @@ type expectedHop struct {
 
 	// Three hop route where fees need to be added in to the forwarding amount.
 	// The high fee hop phamnewun should be avoided.
-	{target: "elst", paymentAmt: 50000, feeLimit: noFeeLimit,
-		expectedTotalTimeLock: 103, expectedTotalAmt: 50050210,
+	{name: "AvoidHighFees", target: "elst",
+		paymentAmt: 50000, pathRestrictions: PathRestrictions{FeeLimit: noFeeLimit, MaxCltvDelay: noCltvLimit},
+		expectedTotalTimeLock: 389, expectedTotalAmt: 50050210,
+		expectedHops: []expectedHop{
+			{alias: "songoku", fwdAmount: 50000200, fee: 50010, timeLock: 245},
+			{alias: "sophon", fwdAmount: 50000000, fee: 200, timeLock: 101},
+			{alias: "elst", fwdAmount: 50000000, fee: 0, timeLock: 101},
+		}},
+	// Force taking the expensive route by setting a maximum cltv delay.
+	// The low fee, longer delay path should be avoided.
+	{name: "AvoidHighCltvPath", target: "elst", paymentAmt: 50000,
+		pathRestrictions:      PathRestrictions{FeeLimit: noFeeLimit, MaxCltvDelay: 10},
+		expectedTotalTimeLock: 255, expectedTotalAmt: 55010220,
 		expectedHops: []expectedHop{
-			{alias: "songoku", fwdAmount: 50000200, fee: 50010, timeLock: 102},
+			{alias: "phamnuwen", fwdAmount: 50000200, fee: 5010020, timeLock: 245},
 			{alias: "sophon", fwdAmount: 50000000, fee: 200, timeLock: 101},
 			{alias: "elst", fwdAmount: 50000000, fee: 0, timeLock: 101},
 		}},
+
 	// Three hop route where fees need to be added in to the forwarding amount.
 	// However this time the fwdAmount becomes too large for the roasbeef <->
 	// songoku channel. Then there is no other option than to choose the
 	// expensive phamnuwen channel. This test case was failing before
 	// the route search was executed backwards.
-	{target: "elst", paymentAmt: 100000, feeLimit: noFeeLimit,
-		expectedTotalTimeLock: 103, expectedTotalAmt: 110010220,
+	{name: "ForceHighFeePath", target: "elst", paymentAmt: 100000,
+		pathRestrictions:      PathRestrictions{FeeLimit: noFeeLimit, MaxCltvDelay: noCltvLimit},
+		expectedTotalTimeLock: 255, expectedTotalAmt: 110010220,
 		expectedHops: []expectedHop{
-			{alias: "phamnuwen", fwdAmount: 100000200, fee: 10010020, timeLock: 102},
+			{alias: "phamnuwen", fwdAmount: 100000200, fee: 10010020, timeLock: 245},
 			{alias: "sophon", fwdAmount: 100000000, fee: 200, timeLock: 101},
 			{alias: "elst", fwdAmount: 100000000, fee: 0, timeLock: 101},
 		}},
 
 	// Basic route with fee limit.
-	{target: "sophon", paymentAmt: 100, feeLimit: 50,
-		expectFailureNoPath: true,
+	{name: "MaxFeeLimit", target: "sophon", paymentAmt: 100,
+		pathRestrictions: PathRestrictions{FeeLimit: 50, MaxCltvDelay: noCltvLimit},
+		expectError:      true, expectedFailure: ErrNoPathFound,
+	},
+
+	// Basic route with max cltv delay limit.
+	{name: "MaxCltvLimit", target: "elst", paymentAmt: 100,
+		pathRestrictions: PathRestrictions{FeeLimit: noFeeLimit, MaxCltvDelay: 9},
+		expectError:      true, expectedFailure: ErrNoPathFound,
+	},
+
+	// Test routing a value for which there is no capacity
```

Comment:
It is a bit hard to see what are the actual test additions here. Ideally the commits would be structured like
1. Modify the test data, any necessary refactoring and existing tests to work with this data.
2. Add the new tests.